Examples/Algorithms/Vertexing/src/IterativeVertexFinderAlgorithm.cpp

```diff
@@ -56,10 +58,31 @@ ActsExamples::IterativeVertexFinderAlgorithm::IterativeVertexFinderAlgorithm(
 ActsExamples::ProcessCode ActsExamples::IterativeVertexFinderAlgorithm::execute(
     const ActsExamples::AlgorithmContext& ctx) const {
   // retrieve input tracks and convert into the expected format
-  const auto& inputTrackParameters =
-      ctx.eventStore.get<TrackParametersContainer>(m_cfg.inputTrackParameters);
-  const auto& inputTrackPointers =
-      makeTrackParametersPointerContainer(inputTrackParameters);
+
+  std::vector<Acts::BoundTrackParameters> inputTrackParameters;
+  std::vector<const Acts::BoundTrackParameters*> inputTrackPointers;
+
+  if (!m_cfg.inputTrackParameters.empty()) {
+    const auto& tmp =
+        ctx.eventStore.get<std::vector<Acts::BoundTrackParameters>>(
+            m_cfg.inputTrackParameters);
+    inputTrackParameters = tmp;
+    inputTrackPointers = makeTrackParametersPointerContainer(tmp);
+  } else {
+    const auto& inputTrajectories =
```

Comment:
I would argue to put parts of this in a utility function in a private header file in `Examples/Algorithms/Vertexing/src`, e.g. like this

```c++
template<typename config_t>
auto makeParameterContainers(const config_t &config, const Acts::Examples::AlgorithmContext &ctx) {
  // ...
  return std::make_tuple(vec, vecPointer);
}

because the string identifier variables are named equally in the three algorithms, right?